web-common/src/features/dashboards/time-dimension-details/charts/patch-vega-spec.ts

```diff
@@ -0,0 +1,151 @@
+import { COMPARIONS_COLORS } from "@rilldata/web-common/features/dashboards/config";
+import {
+  MainAreaColorGradientDark,
+  MainLineColor,
+} from "@rilldata/web-common/features/dashboards/time-series/chart-colors";
+import { TIME_GRAIN } from "@rilldata/web-common/lib/time/config";
+import { V1TimeGrain } from "@rilldata/web-common/runtime-client";
+import { VisualizationSpec } from "svelte-vega";
+import { TDDAlternateCharts, TDDChart } from "../types";
+
+function patchSpecForTimeComparison(
+  sanitizedSpec,
+  chartType: TDDAlternateCharts,
+  timeUnit: string,
+  measureName: string,
+  yEncoding,
+  colorEncoding,
+) {
+  yEncoding.field = "measure";
+
+  sanitizedSpec.transform = [
+    // Sanitize and transform comparison data in the right time format
+    { timeUnit: timeUnit, field: "comparison\\.ts", as: "comparison_ts" },
+    // Expand datum to have a key field to differentiate between current and comparison data
+    { fold: ["ts", "comparison_ts"] },
+    // Add a measure field to hold the right measure value
+    {
+      calculate: `(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])`,
+      as: "measure",
+    },
+    // Add a time field to hold the right time value
+    {
+      calculate:
+        "(datum['key'] === 'comparison_ts' ? datum['comparison_ts'] : datum['ts'])",
+      as: "time",
+    },
+  ];
```

Comment:
**Performance Issue**: The transformation logic within the 'patchSpecForTimeComparison' function might lead to performance issues due to the complex data manipulation operations like folding and conditional calculations within the Vega-Lite specification. These operations can be computationally expensive, especially with large datasets. <br> **Fix**: Optimize the transformation logic by pre-processing the data outside of Vega-Lite if possible, or simplify the transformations to reduce computational overhead. <br> **Code Suggestion**: 
 ```
 -  sanitizedSpec.transform = [
 -    // Sanitize and transform comparison data in the right time format
 -    { timeUnit: timeUnit, field: "comparison\\.ts", as: "comparison_ts" },
 -    // Expand datum to have a key field to differentiate between current and comparison data
 -    { fold: ["ts", "comparison_ts"] },
 -    // Add a measure field to hold the right measure value
 -    {
 -      calculate: '(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])',
 -      as: "measure",
 -    },
 -    // Add a time field to hold the right time value
 -    {
 -      calculate:
 -        "(datum['key'] === 'comparison_ts' ? datum['comparison_ts'] : datum['ts'])",
 -      as: "time",
 -    },
 -  ];
 +  // Consider optimizing or simplifying these transformations
 ```